### PR TITLE
Add a standard function to load credentials from the environment

### PR DESCRIPTION
**Pull request by jhford**
This function is useful to implementors of the library which are to be used on
the command line.  Having a stanard method defined in the main client library
means that we can have identical behaviour across multiple consumers of the
library.  My proposal is to add loading credentials from configuration files.
For all cases, I think that a non-fatal error for this function is the right
action to take, so that if something goes wrong while loading a file, we just
return a nil value Credential.  I'm also happy to have it return an error and
have the caller ignore the error explicitly.  If we do that, we should add an
error return to this function

### CODE REVIEW

File: creds.go [hunk 1]
```diff
@@ -195,3 +196,25 @@ func (creds *Credentials) Cert() (cert *Certificate, err error) {
 	}
 	return
 }
+
+// LoadCredentials attempts to load a set of Taskcluster Credentials from the
+// environment using the standard Taskcluster environment variables.  This
+// function never fails, but might return a pointer to a nil
+// tcclient.Credentials struct.
```
Comment (petemoore) [inline creds.go]: It would be nice to be specific here and mention which environment variables.

I tend to think it would be useful to return an error which can be ignored, if no client ID is present, or no access token is present, or if they don't confirm to a given regex, or if a certificate is provided but can be seen to be invalid from its format. In the case of invalid credentials being in the environment, I think it would still be nice to provide the incomplete/invalid credentials struct back, together with the error, so if e.g. someone wants to make a call to this function, knowing that e.g. TASKCLUSTER_ACCESS_TOKEN is not set, they can still use it to set the other fields, ignore the error, and then set the last field themselves.

To this end it might also be nice to be able to have a separate function:

```go
func (c *Credentials) Validate() error {
    ...
}
```

that validates the credentials, that is called by your new `LoadCredentials` function internally. This way, if someone intends to call it knowing the credentials are invalid, wishes to mutate them, and then validate them, they can reuse the logic to validate them in the `Validate` method, e.g.:

```go
// ignore error here, since we know password is not in environment
creds, _ := tcclient.LoadCredentials()
creds.AccessToken = os.Getenv("SuperSecretAccessToken")
if err := creds.Validate(); err != nil {
    log.Fatalf("Invalid taskcluster credentials: %v", err)
}
```

I think afterwards, we could also apply a change such that when we generate the packages for each taskcluster service, we could adapt the constructors to allow passing a nil `*Credentials` to imply that the credentials should be loaded from the environment. Note this would be subtly different from passing a pointer to an empty Credentials struct, which would be treated as meaning you really want to use no credentials, rather than loading from the environment (e.g. because you are only hitting public endpoints that don't require an `Authorization` header).

In other words:

```go
var q *queue.Queue
q = queue.New(nil) // initialises based on env variables
q = queue.New(&tcclient.Credentials{}) // initialises based on empty credentials
```

This doesn't have to be done in this PR, but I can imagine it would be nice for us to do this after this PR lands, so this is a really nice change.

File: creds.go [hunk 2]
```diff
@@ -195,3 +196,25 @@ func (creds *Credentials) Cert() (cert *Certificate, err error) {
...
+func LoadCredentials() *Credentials {
```
Comment (petemoore) [inline creds.go]: Maybe we should call this `CredentialsFromEnvVars` or something similar, so it is clear credentials are initialised based on environment variables?

File: creds.go [hunk 3]
```diff
@@ -195,3 +196,25 @@ func (creds *Credentials) Cert() (cert *Certificate, err error) {
...
+
+	if value, present := os.LookupEnv("TASKCLUSTER_ACCESS_TOKEN"); present {
+		creds.AccessToken = value
+	}
+
+	if value, present := os.LookupEnv("TASKCLUSTER_CERTIFICATE"); present {
+		creds.Certificate = value
+	}
+
+	return creds
```
Comment (petemoore) [inline creds.go]: This method could be simplified to:

```go
func LoadCredentials() *Credentials {
    return &Credentials{
        ClientID:    os.Getenv("TASKCLUSTER_CLIENT_ID"),
        AccessToken: os.Getenv("TASKCLUSTER_ACCESS_TOKEN"),
        Certificate: os.Getenv("TASKCLUSTER_CERTIFICATE"),
    }
}
```

since if an env var isn't set, the resulting value would be the zero value, which is an empty string.

**Review by petemoore — CHANGES_REQUESTED** [review]
Thanks John for doing this! I'd never got round to doing it, but had it at the back of my mind, so great you went ahead and did it!

**Review by jhford — COMMENTED** [review]
This is much better than my initial implementation!  I really like this.
